File: c/12.c
// Highly divisible triangular number

#include <stdio.h>
#define MAX 500

int triangle(int max) {
	int sum = 0;
	for (int i=1; i<=max; ++i) {
		sum += i;
	} return sum;
}

void printFactors(int num) {
	for (int i=1; i<=num; ++i) {
		if (num%i==0) printf("%d,", i);
	}
}

int amountOfFactors(int num) {
	int amount = 0;
	for (int i=1; i<=num; ++i) {
		if (num%i==0) amount++;
	} return amount;
}

int main(int argc, char* argv[]) {
	
	int tri, factors = 0;
	int i = 58996953;  
	while (factors <= MAX) {
		
		tri =  triangle(i);
		factors = amountOfFactors(tri);
		printf("%d: %d factors\n", tri, factors);
		//printFactors(tri); printf("\n");
		i++;
	}

	return 0;
}

File: js/12.js
function isInt(n) {
	return (n%1==0)? true: false;
}

function factors(n) {
	let out = [];
	for (var i=1; i<=n; i++) {
		if (isInt(n/i)) out.push(i)
	}
	return out;
}

// function tri(n) {
// 	let out = 0;
// 	for (var i=1; i<=n; i++) {
// 		out += i;
// 	}
// 	return out;
// }

function highlyDivisibleTriangle(amountOfFactors) {
	let found = false;
	let i = 1;
	let tri = 1;
	while (!found) {
		let f = factors(tri);
		console.log(tri+": "+f.length+" factors");
		if (f.length >= amountOfFactors) {
			console.log("done!");
			found = true;
			break;
		}
		i++;
		tri += i;
	}
}

highlyDivisibleTriangle(500);